BASE/src/Microsoft.ApplicationInsights/TelemetryClient.cs

```diff
@@ -83,8 +83,10 @@ public string InstrumentationKey
 
         /// <summary>
         /// Gets the <see cref="TelemetryConfiguration"/> object associated with this telemetry client instance.
+        /// This property is an internal part of Application Insights infrastructure. Changes made to the configuration will not be applied in. Do not use.
         /// </summary>
-        internal TelemetryConfiguration TelemetryConfiguration
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public TelemetryConfiguration TelemetryConfiguration
```

Comment:
I am not sure of the part - "Changes made to the configuration will not be applied in". If one modifies the config, I think it would get applied to all telemetry tracker afterwards.

Reply:
You are right, I misread [this comment](https://github.com/microsoft/ApplicationInsights-dotnet/issues/581#issuecomment-313280554). Should we instead warn that changes can affect other clients or is that not longer valid given the date of the original comment?

Reply:
I'd expect it would affect other clients made from the same config.

btw - what is your use case for accessing Configuration from client?

Reply:
Also yes to the question -  we need to add this warning. Deprecating .Active wouldn't impact this.

Reply:
@TimothyMothra @rajkumar-rangaraj Do you see any issues making configuration public, but hidden to avoid misuse.?

Reply:
I reviewed this PR earlier and feel it is hidden to avoid the misuse. I don't see any problem with making this public as we give control to TelemetryConfiguration in other places.